### ISSUE TITLE
idea: make deletion a harder process

### ISSUE THREAD

**Issue by dankelley** [dankelley/nb#20]
I don't like the idea that hitting a ``d`` instead of an ``e`` will delete a note. (I've done that!)

I have a couple of ideas.
1. require the full word, i.e. ``nb --delete hash`` instead of ``nb -d hash``
2. make ``--delete`` only mark the note as "trash" and require a separate operation to clear the trash

PS. there is really very little memory cost associated with keeping notes in a trashcan. This is mainly just text, and disks are designed to store movies nowadays. Example: OAR is only 1 Mb of characters, so it would take 1000 books to make the database get to the RAM of even a small machine.

**Comment by richardsc** [comment]
I like the idea of a trash can. Perhaps the number of notes in the "trash" can be displayed when a user looks at the full list, 
```
$ nb
a1: foo
b2: bar

2 trashed notes
```
and the actual contents can be listed by something like:
```
$ nb --trash
c3: deleted note
```
and emptied with:
```
$ nb --emptytrash
2 notes purged from trash
```

This means that there should also be an `undelete` option, to get things back from the trash:
```
$ nb -u c3
Restored note c3 from trash
```

**Comment by dankelley** [comment]
I really like your suggested syntax.

Any thoughts on removing the ``-d`` and requiring ``--delete``?

**Comment by richardsc** [comment]
Sorry, forgot to comment on the `-d` vs `--delete`. I kind of like the `-d` syntax, if there is going to be a trash. But I'm lazy ... Seems like if you *have* to type `--emptytrash` to permanently remove a note, there's not nearly as much danger in using the `-d` shortcut.

**Comment by dankelley** [comment]
Just put ``-d`` back. Logging out for day.

**Comment by dankelley** [comment]
I've just added ``--emptytrash`` ... in ``nota`` and back-ported to ``nb`` (both ``master`` and ``develop`` branches in each of these).

Whew. Having two repos and two branches multiplies the work by a lot!

PS. I have switched entirely to ``nota``. However, I do not know if that's to be recommended yet, because at some point (Saturday probably) I'll try to make this into an official python package and that may cause code to stop working until I can figure out the directory structure etc. It's kind of complicated, actually, and I wish I figured it out 10 years ago, because I think asking people to copy files to directories manually is not good.
